### PR TITLE
Add Case, NewCase request/reponse defs to curator service

### PR DESCRIPTION
**Pull request by khmoran**

### CODE REVIEW

**Comment by khmoran** [comment]
[S#7: Design & implement 'Case' API entity in curator service](https://trello.com/c/gnnb3fDQ/290-s7-design-implement-case-api-entity-in-curator-service)

File: verification/curator-service/api/openapi/openapi.yaml [hunk 1]
```diff
@@ -500,11 +503,154 @@ components:
...
+                type: string
+            occupation:
+              type: string
+        location:
+          $ref: "#/components/schemas/Location"
+        events:
+          description: > 
+            An event with name "confirmed" must be specified. Other event names
+            include onsetSymptoms, firstClinicalConsultation, selfIsolation,
+            hospitalAdmission, icuAdmission, and outcome.
```
Comment (axmb) [inline verification/curator-service/api/openapi/openapi.yaml:549]: Not a change to make now, but if we're dictating the title of events, I wonder if the `name` should be an enum. Or for lack of that, if this is the best place to document the "standard" value set.

File: verification/curator-service/api/openapi/openapi.yaml [hunk 2]
```diff
@@ -522,9 +689,27 @@ components:
...
+        newly-created cases.
+      allOf:
+        - $ref: "#/components/schemas/Case"
+      properties:
+        caseReference:
+          required:
+            - sourceId
+            - sourceUrl
+        location:
+          description: Note that either `geometry` or `query` is required.
```
Comment (axmb) [inline verification/curator-service/api/openapi/openapi.yaml:703]: Isn't it `query || (geometry && country && geoResolution)`?

File: verification/curator-service/api/openapi/openapi.yaml (hunk 1 above)
Comment (khmoran) [reply verification/curator-service/api/openapi/openapi.yaml:550]: yeah enum probably makes sense -- add a todo

File: verification/curator-service/api/openapi/openapi.yaml (hunk 2 above)
Comment (khmoran) [reply verification/curator-service/api/openapi/openapi.yaml]: yes, thanks!

**Pull request merged**